plutus-core/plutus-core/src/PlutusCore/Default/Builtins.hs

```diff
@@ -360,6 +520,22 @@ instance Flat DefaultFun where
               Nop1                 -> 25
               Nop2                 -> 26
               Nop3                 -> 27
+              Fst                  -> 31
```

Comment:
We very much don't want any automation in encoding, 'cause automated stuff can change arbitrarily and we want encoding/decoding to be stable. However we could derive decoding in terms of encoding using that [`inverseMap`](https://kowainik.github.io/posts/haskell-mini-patterns#bidirectional-parsing) (in our case it's an array) trick. This would also save us from this problem when the programmer updates the encoding part by adding a new builtin and forgets to update the decoding part and doesn't get a warning (but tests do fail). However that is probably going to make decoding a tiny tiny bit less efficient (pattern matching vs array lookup). @michaelpj thoughts?

Reply:
We could do that. It might be fine. If we're going to say "fun should be representable as an Int" we could also use that here.

Reply:
I think regardless of how we represent `fun`, encoding stuff should be written out explicitly (and deriving decoding in terms of encoding is fine regardless of how `fun` is represented).